frontend/packages/console-app/src/components/data-view/useConsoleDataViewFilters.ts

```diff
@@ -10,17 +10,19 @@ import {
 import { getLabelsAsString } from '@console/shared/src/utils/label-filter';
 import { ResourceFilters } from './types';
 
-export const useResourceDataViewFilters = <
-  TData extends K8sResourceCommon = K8sResourceCommon,
+export const useConsoleDataViewFilters = <
+  TData,
   TFilters extends ResourceFilters = ResourceFilters
 >({
   data,
   initialFilters,
+  getObjectName,
   matchesAdditionalFilters,
 }: {
   data: TData[];
   initialFilters: TFilters;
-  matchesAdditionalFilters?: (resource: TData, filters: TFilters) => boolean;
+  getObjectName?: (obj: TData) => string;
```

Comment:
I think this we should comment this behaviour because it's not obvious

```suggestion
  /** 
   * Override the default getter function for the name of a given `TData`.
   * 
   * Defaults to assuming that `TData` can be coerced to {@link K8sResourceCommon}.
   */
  getObjectName?: (obj: TData) => string;
```